Modules/Messages/src/Messages.Application/Messages/Commands/SendMessage/Handler.cs

```diff
@@ -59,14 +71,13 @@ public async Task<SendMessageResponse> Handle(SendMessageCommand request, Cancel
 
     private async Task SaveMessage(Message message, CancellationToken cancellationToken)
     {
-        await _dbContext.Set<Message>().AddAsync(message, cancellationToken);
-        _blobStorage.Add(_blobOptions.RootFolder, message.Id, message.Body);
+        await _messagesRepository.Add(message, cancellationToken);
 
+        _blobStorage.Add(_blobOptions.RootFolder, message.Id, message.Body);
```

Comment:
The MessagesRepository should be responsible for saving the content. That's one of the main reasons for this whole refactoring. Because at the moment, the Command and Query handlers need to know way to much about how entities are persisted. 

So you should move the following line:

```
_blobStorage.Add(_blobOptions.RootFolder, message.Id, message.Body);
```

to the `MessagesRepository`. Also, remove the `IBlobStorage` dependency from the constructor of this class.

(I just realized that I wasn't explicit about that point in the task description and updated it accordingly)